common/browser.go

```diff
@@ -128,13 +127,23 @@ func newBrowser(
 func (b *Browser) connect() error {
 	b.logger.Debugf("Browser:connect", "wsURL:%q", b.browserProc.WsURL())
 
+	// The browser needs its own context to correctly close dependencies such
+	// as the connection. It cannot rely on the vuCtx since that would close
+	// the connection too early. The connection and subprocess need to be
+	// shutdown at around the same time to allow for any last minute CDP
+	// cleanup messages to be sent to chromium.
+	b.internalCtx, b.InternalCancelFn = context.WithCancel(context.Background())
```

Comment:
So `NewConnection`'s method signature would become:

```go
func NewBrowser(
        browserCtx context.Context,
	vuCtx context.Context,
	vuCtxCancelFn context.CancelFunc,
	browserProc *BrowserProcess,
	browserOpts *BrowserOptions,
	logger *log.Logger,
)
```

And now [Launch](https://github.com/grafana/xk6-browser/blob/e24c1a5b448cf982bd9d4445b074717d0c54eef3/chromium/browser_type.go#L217) and [Connect](https://github.com/grafana/xk6-browser/blob/e24c1a5b448cf982bd9d4445b074717d0c54eef3/chromium/browser_type.go#L136) would create a `background` context. Is that how you envision it?

Reply:
> This will become buggy later on since we have a lot of contexts now.

Yeah, this is why i think `browser` needs a bit of work to separate some responsibilities -- for another day.

Reply:
Ideally, I imagined it as passing a `Context` from the top level, maybe from `RootModule.NewModuleInstance`.

Reply:
I've added it back in so now there are two contexts in `browser`: 1243f0688629b6df8741c9bf509894359c908135.

WDYT?